### PR TITLE
Add static HTML deploy mode

### PR DESCRIPTION
**Pull request by indytechcook**
## Description

See  https://github.com/department-of-veterans-affairs/va.gov-cms/issues/2851.

## Testing done

Did a screenshare with @swirtSJW and @ElijahLynn.

## Definition of Done
- [ ] Product release are written (or in progress), if required.
- [ ] Documentation has been updated, if applicable.

---
This change is [Reviewable](https://reviewable.io/reviews/department-of-veterans-affairs/va.gov-cms/2878)

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/src/Deploy/Plugin/CMSExport.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Drupal\va_gov_backend\Deploy\Plugin;
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/src/Deploy/Plugin/CMSExport.php:3]: Just noting this is not being used right now but will be in another PR.

File: docroot/sites/default/settings.php [hunk 2]
```diff
@@ -169,6 +169,11 @@
   include $app_root . '/' . $site_path . '/settings/settings.fast_404.php';
 }
 
+// Deploy Settings
```
Comment (swirtSJW) [inline docroot/sites/default/settings.php:172]: I'd like to see some explanation here like...
```suggestion
// Jenkins moves this file into place during deploy, so if it is present we are in deploy mode.
```

File: docroot/sites/default/settings.php [hunk 3]
```diff
@@ -184,5 +189,12 @@
   include $app_root . '/' . $site_path . '/settings/settings.local.php';
 }
 
+// The VA_GOV_IN_DEPLOY_MODE is set in settings.deploy.active.php.
```
Comment (swirtSJW) [inline docroot/sites/default/settings.php:192]: Could all this reside higher so that at lives with lines 173-176 so it reads nicer and is clearer?

File: docroot/sites/default/settings/settings.deploy.inactive.php [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?php
+
```
Comment (swirtSJW) [inline docroot/sites/default/settings/settings.deploy.inactive.php:2]:
```suggestion
// Ansible moves this file around to make deploy mode active or inactive.
```

**Review by swirtSJW — DISMISSED** [review]
Just some minor code comment suggestions.   I really like this architecture Neil.  Well done.

File: docroot/sites/default/settings.php (hunk 3 above)
Comment (indytechcook) [reply docroot/sites/default/settings.php:192]: I wanted it after the `settings.local.php` so I could test without actually renaming the file.

**Comment by indytechcook** [comment]
Matching Devops PR: https://github.com/department-of-veterans-affairs/devops/pull/7433

**Review by swirtSJW — APPROVED** [review]
